libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs

```diff
@@ -405,6 +405,27 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                 // return the actionResult as the result of this operation
                 return await dc.EndDialogAsync(result: requestResult, cancellationToken: cancellationToken).ConfigureAwait(false);
             }
+            catch (HttpRequestException ex)
+            {
+                var requestResult = new Result()
+                {
+                    StatusCode = (int)HttpStatusCode.NotFound,
```

Comment:
why not use the status code from the HttpRequestException itself?

Reply:
Unfortunately, the inner SocketException does not translate into an HttpStatusCode, which is currently expected for Result:

![image](https://user-images.githubusercontent.com/11055362/177667273-b5102ad2-3b92-49fa-bc1f-bdd7d00af9c4.png)

Not sure how to further extend this to account for this exception other than hard coding 'NotFound'. EnsureSucessStatusCode doesn't seem valid here. We could maybe provide more info in the trace. 

Thoughts?

Reply:
yes makes sense - this isn't an http error it's at the network connection level

Reply:
nit: make a comment in the code?
we just don't want a future maintainer of this code getting confused and thinking this was an actual response